python/common/org/python/types/Slice.java

```diff
@@ -240,4 +240,85 @@ public ValidatedValue validateValueTypes() {
             return org.python.types.NotImplementedType.NOT_IMPLEMENTED;
         }
     }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"index"}
+    )
+    public org.python.Object __getitem__(org.python.Object index) {
+        throw new org.python.exceptions.TypeError("'slice' object is not subscriptable");
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"index", "value"}
+    )
+    public void __setitem__(org.python.Object index, org.python.Object value) {
+        throw new org.python.exceptions.TypeError("'slice' object does not support item assignment");
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"index"}
+    )
+    public void __delitem__(org.python.Object index) {
+        throw new org.python.exceptions.TypeError("'slice' object does not support item deletion");
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"other"}
+    )
+    public org.python.Object __mul__(org.python.Object other) {
+        if (other instanceof org.python.types.ByteArray || other instanceof org.python.types.Bytes ||
+                other instanceof org.python.types.List || other instanceof org.python.types.Str ||
+                other instanceof org.python.types.Tuple) {
+            throw new org.python.exceptions.TypeError("can't multiply sequence by non-int of type 'slice'");
+        } else {
+            return super.__mul__(other);
+        }
+    }
+
+    @org.python.Method(
+            __doc__ = "",
+            args = {"other"}
+    )
+    public org.python.Object __imul__(org.python.Object other) {
+        if (other instanceof org.python.types.ByteArray || other instanceof org.python.types.Bytes ||
+                other instanceof org.python.types.List || other instanceof org.python.types.Str ||
+                other instanceof org.python.types.Tuple) {
```

Comment:
Could you refactor this check into a static method `isSequence`, and then reuse the it in `__mul__` ?